code/modules/antagonists/eldritch_cult/eldritch_structures.dm

```diff
@@ -0,0 +1,161 @@
+/obj/structure/eldritch_crucible
+	name = "Mawed Crucible"
+	desc = "Immortalized cast iron, the steel-like teeth holding it in place, it's vile extract has the power of rebirthing things, remaking them from the very beginning."
+	icon = 'icons/obj/eldritch.dmi'
+	icon_state = "crucible"
+	anchored = FALSE
+	density = TRUE
+	var/current_mass = 5
+	var/max_mass = 5
+	var/in_use = FALSE
+
+/obj/structure/eldritch_crucible/examine(mob/user)
+	. = ..()
+	if(!IS_HERETIC(user) && !IS_HERETIC_MONSTER(user))
+		return
+	if(current_mass < max_mass)
+		. += "The Crucible requires [max_mass - current_mass] more organs or bodyparts!"
+	else
+		. += "The Crucible is ready to be used!"
+
+	. += "You can anchor and reanchor it using Codex Cicatrix!"
+	. += "It is currently [anchored == FALSE ? "unanchored" : "anchored"]"
+
+/obj/structure/eldritch_crucible/attacked_by(obj/item/I, mob/living/user)
+	if(istype(I,/obj/item/nullrod))
+		qdel(src)
+		return
+
+	if(!IS_HERETIC(user) && !IS_HERETIC_MONSTER(user))
+		if(iscarbon(user))
+			devour(user)
+		return
+
+	if(istype(I,/obj/item/forbidden_book))
+		playsound(src, 'sound/misc/desecration-02.ogg', 75, TRUE)
+		anchored = !anchored
+		to_chat(user,"<span class='notice'>You [anchored == FALSE ? "unanchor" : "anchor"] the crucible</span>")
+		return
+
+	if(istype(I,/obj/item/bodypart) || istype(I,/obj/item/organ))
+		if(current_mass >= max_mass)
+			to_chat(user,"<span class='notice'> Crucible is already full!</span>")
+			return
+		playsound(src, 'sound/items/eatfood.ogg', 100, TRUE)
+		to_chat(user,"<span class='notice'>Crucible devours [I.name] and fills itself with a little bit of liquid!</span>")
+		current_mass++
+		qdel(I)
+		update_icon_state()
+		return
+
+	return ..()
+
+/obj/structure/eldritch_crucible/attack_hand(mob/user)
+	if(!IS_HERETIC(user) && !IS_HERETIC_MONSTER(user))
+		if(iscarbon(user))
+			devour(user)
+		return
+
+	if(in_use)
+		to_chat(user,"<span class='notice'>Crucible is already in use!</span>")
+		return
+
+	if(current_mass < max_mass)
+		to_chat(user,"<span class='notice'>Crucible isn't full! Bring it more organs or bodyparts!</span>")
+		return
+
+	in_use = TRUE
+	var/list/lst = list()
+	for(var/X in subtypesof(/obj/item/eldritch_potion))
+		var/obj/item/eldritch_potion/potion = X
+		lst[initial(potion.name)] = potion
+	var/type = lst[input(user,"Choose your brew","Brew") in lst]
+	playsound(src, 'sound/misc/desecration-02.ogg', 75, TRUE)
+	new type(drop_location())
+	current_mass = 0
+	in_use = FALSE
+	update_icon_state()
+
+/obj/structure/eldritch_crucible/proc/devour(mob/living/carbon/user)
+	playsound(src, 'sound/items/eatfood.ogg', 100, TRUE)
+	to_chat(user,"<span class='danger'>Crucible grabs your arm and devours it whole!</span>")
+	var/obj/item/bodypart/arm = user.get_active_hand()
+	arm.dismember()
+	qdel(arm)
+	current_mass += current_mass < max_mass ? 1 : 0
```

Comment:
imo, it should just not delete their arm if it's full (it can still dismember it, though).

then again, that might dampen the meme potential of just leaving this thing in a main hallway and watching gamers intentionally sac limbs to it

Reply:
yeah it will dampen the meme potential